model/role.go

```diff
@@ -851,6 +864,7 @@ func MakeDefaultRoles() map[string]*Role {
 			PermissionCreateDirectChannel.Id,
 			PermissionCreateGroupChannel.Id,
 			PermissionViewMembers.Id,
+			PermissionCreateTeam.Id,
 		},
 		SchemeManaged: true,
 		BuiltIn:       true,
```

Comment:
What was the reason for the default roles getting added permissions?

Reply:
The old configs were used to add more permissions to each role on app setup (utils/authorization.go). Those configs are now gone so we don't have any way to add those permissions on startup. I added the permissions to the appropriate roles based on their old config default value.

Reply:
So if I'm understanding correctly the permissions that are added here to the defaults are from the `else` cases in https://github.com/mattermost/mattermost-server/pull/18035/files#diff-f8055deb0d618974840639d595f837567ad315388c8cbdefe56b19c00b6ec22f if no less restricting config setting was found?

Reply:
Yes they're from the `else` cases